File: front/src/types/article.ts
import type { User } from './user'

export interface Tag {
  id: number
  name: string
  articleCount?: number
  createdBy?: number
}

export interface ArticleListItem {
  id: number
  title: string
  summary: string
  aiSummary?: string
  aiSummaryStatus?: number
  aiSummaryLong?: string
  aiSummaryLongStatus?: number
  coverImage?: string | null
  author: {
    id: number
    username: string
    avatar?: string | null
  }
  tags: Tag[]
  viewCount: number
  likeCount: number
  commentCount: number
  createdAt: string
  updatedAt?: string
  status?: string
}

export interface Article {
  id: number
  title: string
  content: string
  summary: string
  coverImage?: string | null
  status: 'PUBLISHED' | 'DRAFT' | 'REVIEWING' | 'REJECTED' | 'APPROVED'
  author: {
    id: number
    username: string
    avatar?: string | null
  }
  tags: Tag[]
  viewCount: number
  likeCount: number
  isLiked?: boolean
  commentCount?: number
  aiSummary?: string
  aiSummaryStatus?: number
  aiSummaryLong?: string
  aiSummaryLongStatus?: number
  createdAt: string
  updatedAt?: string
}

export interface HotArticle {
  rank: number
  id: number
  title: string
  summary: string
  viewCount: number
  likeCount: number
  author: {
    id: number
    username: string
    avatar?: string | null
  }
  createdAt: string
}

export interface ArticleQuery {
  page?: number
  pageSize?: number
  tagId?: number
  tagName?: string
  orderBy?: 'latest' | 'hot' | 'oldest'
  authorId?: number
}

export interface ArticleFormData {
  title: string
  content: string
  summary?: string
  coverImage?: string
  tagIds?: number[]
  tagNames?: string[]
  status?: 'PUBLISHED' | 'DRAFT'
  genAiSummary?: 0 | 1
  genAiSummaryLong?: 0 | 1
}

File: blog-backend-go/internal/dto/article.go
package dto

import "time"

// ArticleCreateRequest is used for creating articles.
type ArticleCreateRequest struct {
	Title            string   `json:"title" binding:"required,max=50"`
	Content          string   `json:"content" binding:"max=10000"`
	Summary          string   `json:"summary" binding:"max=300"`
	CoverImage       string   `json:"coverImage"`
	TagIDs           []int64  `json:"tagIds" binding:"max=10"`
	TagNames         []string `json:"tagNames" binding:"max=10"`
	Status           string   `json:"status"`
	GenAiSummary     int      `json:"genAiSummary"`
	GenAiSummaryLong int      `json:"genAiSummaryLong"`
}

// ArticleUpdateRequest is used for updating articles.
type ArticleUpdateRequest = ArticleCreateRequest

// ArticleQuery holds query parameters for article list.
type ArticleQuery struct {
	Page     int    `form:"page,default=1"`
	PageSize int    `form:"pageSize,default=10"`
	TagName  string `form:"tagName"`
	AuthorID int64  `form:"authorId"`
	OrderBy  string `form:"orderBy,default=latest"`
	Status   string `form:"status"`
	Keyword  string `form:"keyword"`
}

// HotArticleQuery holds query parameters for hot articles.
type HotArticleQuery struct {
	Limit int `form:"limit,default=10"`
}

// SearchArticleQuery holds query parameters for article search.
type SearchArticleQuery struct {
	Keyword  string `form:"keyword" binding:"required"`
	Page     int    `form:"page,default=1"`
	PageSize int    `form:"pageSize,default=10"`
}

// ArticleVO is the article response.
type ArticleVO struct {
	ID                int64     `json:"id"`
	Title             string    `json:"title"`
	Content           string    `json:"content,omitempty"`
	Summary           string    `json:"summary"`
	CoverImage        string    `json:"coverImage"`
	Status            string    `json:"status"`
	Author            *AuthorVO `json:"author"`
	Tags              []TagVO   `json:"tags"`
	ViewCount         int64     `json:"viewCount"`
	LikeCount         int64     `json:"likeCount"`
	CommentCount      int       `json:"commentCount"`
	IsLiked           bool      `json:"isLiked"`
	CreatedAt         time.Time `json:"createdAt"`
	UpdatedAt         time.Time `json:"updatedAt"`
	AiSummary         string    `json:"aiSummary"`
	AiSummaryStatus   int       `json:"aiSummaryStatus"`
	AiSummaryLong     string    `json:"aiSummaryLong"`
	AiSummaryLongStatus int     `json:"aiSummaryLongStatus"`
}

// AiSummaryLongVO is the long AI summary response.
type AiSummaryLongVO struct {
	Status     int    `json:"status"`
	SummaryLong string `json:"summaryLong"`
}
